src/components/Modal/index.tsx

```diff
@@ -0,0 +1,102 @@
+import {
+    memo,
+    useRef,
+    useState,
+    forwardRef,
+    useImperativeHandle,
+} from 'react';
+
+import { useOutsideClick } from '../../hooks';
+import { Close as CloseIcon } from '../../icons';
+import { Button } from '../Button/Button';
+import Styled from './styles';
+import type { ModalProps, ModalBaseProps } from './types';
+
+const Component: React.ForwardRefRenderFunction<ModalProps, ModalBaseProps> = ({
+    clear,
+    close = true,
+    title,
+    shadow = true,
+    actions,
+    children,
+    initialValue = false,
+    outsideClick = true,
+    ...rest
+}, ref) => {
+    const [show, onShow] = useState(initialValue);
+
+    const wrapper = useRef<HTMLDivElement>(null);
+
+    const opened = () => onShow(true);
+
+    const closed = () => onShow(false);
+
+    const outside = () => {
+        closed();
+
+        if (clear) clear();
+    };
+
+    useImperativeHandle(ref, () => ({
+        opened,
+        closed,
+    }));
+
+    useOutsideClick(wrapper, () => outsideClick && outside());
```

Comment:
I think all of these functions can be wrapped in useCallback to prevent re-rendering.

Reply:
On this point it doesn't make much sense to use useCallback as they are very simple functions

Therefore, it is easier to be re-rendered, than to spend memory to check if the function has mutated.

What's your opinion about?

Reply:
Yes actually, you are correct. Too small of functions.